src/ameba/formatter/github_actions_formatter.cr

```diff
@@ -0,0 +1,71 @@
+require "./util"
+
+module Ameba::Formatter
+  # A formatter that outputs issues in a GitHub Actions compatible format.
+  #
+  # See [GitHub Actions documentation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions) for details.
+  class GitHubActionsFormatter < BaseFormatter
+    @mutex = Mutex.new
+
+    # Reports a result of the inspection of a corresponding source.
+    def source_finished(source : Source) : Nil
+      source.issues.each do |issue|
+        next if issue.disabled?
+
+        @mutex.synchronize do
+          output << "::"
+          output << command_name(issue.rule.severity)
+          output << " "
+          output << "file="
+          output << escape_property(source.path)
+          if location = issue.location
+            output << ",line="
+            output << location.line_number
+            output << ",col="
+            output << location.column_number
+          end
+          if end_location = issue.end_location
+            output << ",endLine="
+            output << end_location.line_number
+            output << ",endColumn="
+            output << end_location.column_number
+          end
+          output << ",title="
+          output << escape_property(issue.rule.name)
+          output << "::"
+          output << escape_data(issue.message)
+          output << "\n"
+        end
+      end
+    end
+
+    private def command_name(severity : Severity) : String
+      case severity
+      in .error?      then "error"
+      in .warning?    then "warning"
+      in .convention? then "notice"
+      end
+    end
+
+    # See for details:
+    # - https://github.com/actions/toolkit/blob/74906bea83a0dbf6aaba2d00b732deb0c3aefd2d/packages/core/src/command.ts#L92-L97
+    # - https://github.com/actions/toolkit/issues/193
+    private def escape_data(string : String) : String
+      string
+        .gsub('%', "%25")
+        .gsub('\r', "%0D")
+        .gsub('\n', "%0A")
+    end
```

Comment:
shouldn't we escape a semicolor here.  Example:

```
$ cat actions-formatter.cr                                                                                                                                                             
begin
rescue Exception
rescue Foo::Exception
end

$ ./bin/ameba actions-formatter.cr -f github-actions                                                                                                                                     
::warning file=actions-formatter.cr,line=3,col=8,endLine=3,endColumn=21,title=Lint/ShadowedException::Shadowed exception found: Foo::Exception
```

Reply:
Nope. See the linked sources for the reference.

Reply:
Looks like it's working:

<img width="856" alt="image" src="https://github.com/user-attachments/assets/c3c07468-5b5d-4e19-90b4-c041d50bc04a">